Lib/hmac.py

```diff
@@ -200,19 +228,22 @@ def digest(key, msg, digest):
         except _hashopenssl.UnsupportedDigestmodError:
             pass
 
-    if callable(digest):
-        digest_cons = digest
-    elif isinstance(digest, str):
-        digest_cons = lambda d=b'': _hashlib.new(digest, d)
-    else:
-        digest_cons = lambda d=b'': digest.new(d)
+    if _hmac is not None and isinstance(digest, str):
```

Comment:
Above you check with just `_hmac`:
```suggestion
    if _hmac and isinstance(digest, str):
```

Reply:
Indeed! Thank you (I'll also remove the is not None for _hashopenssl)